New TF examples

Rocketknight1 opened the pull request:
Opening a PR to get some feedback on the new TF example style before I write the rest.

Don't merge it yet, I haven't even finalized the filenames!

fghuman committed:
Added documentation for data collator. (#10941)

* Added documentation for data collator.

* Update docs/source/data_collator.rst

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Added documentation for data collator.

* Added documentation for the data collator.

* Merge branch 'doc_DataCollator' of C:\Users\mahii\PycharmProjects\transformers with conflicts.

* Update documentation for the data collator.

* Update documentation for the data collator.

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>
Co-authored-by: Amna <A.A.Ahmad@student.tudelft.nl>

tma15 committed:
Fix typo (#11188)

NielsRogge committed:
Add DeiT (PyTorch) (#11056)

* First draft of deit

* More improvements

* Remove DeiTTokenizerFast from init

* Conversion script works

* Add DeiT to ViT conversion script

* Add tests, add head model, add support for deit in vit conversion script

* Update model checkpoint names

* Update image_mean and image_std, set resample to bicubic

* Improve docs

* Docs improvements

* Add DeiTForImageClassificationWithTeacher to init

* Address comments by @sgugger

* Improve feature extractors

* Make fix-copies

* Minor fixes

* Address comments by @patil-suraj

* All models uploaded

* Fix tests

* Remove labels argument from DeiTForImageClassificationWithTeacher

* Fix-copies, style and quality

* Fix tests

* Fix typo

* Multiple docs improvements

* More docs fixes

cronoik committed:
Replaced `which` with `who` (#11183)

LysandreJik committed:
Import torch.utils.checkpoint in ProphetNet (#11214)

philschmid committed:
Sagemaker test docs update for framework upgrade (#11206)

* increased train_runtime for model parallelism

* added documentation for framework upgrade

calpt committed:
Use MSELoss in (M)BartForSequenceClassification (#11178)

cceyda committed:
wav2vec2 converter: create the proper vocab.json while converting fairseq wav2vec2 finetuned model (#11041)

* add vocab while converting wav2vec2 original finetuned model

* check save directory exists

* return_attention_mask fix

* quality

LysandreJik committed:
Add Matt as the TensorFlow reference (#11212)

LysandreJik committed:
Fix GPT-2 warnings (#11213)

* Fix GPT-2 warnings

* Update src/transformers/models/gpt2/modeling_gpt2.py

Co-authored-by: Stas Bekman <stas00@users.noreply.github.com>

Co-authored-by: Stas Bekman <stas00@users.noreply.github.com>

patil-suraj committed:
fix docstrings (#11221)

forest1988 committed:
Add documentation for BertJapanese (#11219)

* Start writing BERT-Japanese doc

* Fix typo, Update toctree

* Modify model file to use comment for document, Add examples

* Clean bert_japanese by make style

* Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Split a big code block into two

* Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Add prefix >>> to all lines in code blocks

* Clean bert_japanese by make fixup

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

sgugger committed:
Replace error by warning when loading an architecture in another (#11207)

* Replace error by warning when loading an architecture in another

* Style

* Style again

* Add a test

* Adapt old test

sgugger committed:
Document v4.5.1

patil-suraj committed:
Refactor GPT2 (#11225)

* refactor GPT2

* fix mlp and head pruning

* address Sylvains comments

* apply suggestion from code review

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

sgugger committed:
Doc check: a bit of clean up (#11224)

philschmid committed:
added cache_dir=model_args.cache_dir to all example with cache_dir arg (#11220)

sgugger committed:
Avoid using no_sync on SageMaker DP (#11229)

sgugger committed:
Indent code block in the documentation (#11233)

* Indent code block

* Indent code blocks version 2

* Quality

LysandreJik committed:
Run CI on deepspeed and fairscale (#11172)

* Run CI on deepspeed and fairscale

* Test it on this branch :)

* Rename

* Update the CI image

stas00 committed:
[Deepspeed] zero3 tests band aid (#11235)

* temp band-aid

* style

Nithin-Holla committed:
Save the Wav2Vec2 processor before training starts (#10910)

Co-authored-by: nithin19 <nithin@amberscript.com>

jstremme committed:
make embeddings plural in warning message (#11228)

LysandreJik committed:
Stale bot updated (#10562)

* Updated stale bot

* Specify issue number

* Remove particular handling of assignees

* Unleash the stalebot

* Remove debug branch

parakalan committed:
Close open files to suppress ResourceWarning (#11240)

Co-authored-by: Sudharsan Thirumalai <sudharsan.t@sprinklr.com>

odellus committed:
Fix dimention misspellings. (#11238)

* Update modeling_gpt_neo.py

dimention -> dimension

* Update configuration_speech_to_text.py

dimention -> dimension

forest1988 committed:
Add prefix to examples in model_doc rst (#11226)

* Add prefix to examples in model_doc rst

* Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

stas00 committed:
[troubleshooting] add 2 points of reference to the offline mode (#11236)

* add 2 points of reference to the offline mode

* link the new doc

* add error message

* Update src/transformers/modeling_utils.py

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* style

* rename

* Trigger CI

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

sgugger committed:
Fix #10128 (#11248)

Rocketknight1 committed:
Trying a new style of TF example

stas00 committed:
[deepspeed] test on one node 2 gpus max (#11237)

* test on one node 2 gpus max

* fix the other place

* refactor

* fix

* cleanup

* more exact version

LysandreJik commented on examples/text-classification/new_tf_text_classification.py:
```diff
@@ -0,0 +1,498 @@
+
+# region Helper classes
+class DataSequence(tf.keras.utils.Sequence):
+    # We use a Sequence object to load the data. Although it's completely possible to load your data as Numpy/TF arrays
+    # and pass those straight to the Model, this constrains you in a couple of ways. Most notably, it requires all
+    # the data to be padded to the length of the longest input example, and it also requires the whole dataset to be
+    # loaded into memory. If these aren't major problems for you, you can skip the sequence object in your own code!
```
Great comment!

Rocketknight1 replied:
Thanks!

LysandreJik commented on examples/text-classification/new_tf_text_classification.py:
```diff
@@ -0,0 +1,498 @@
+                     if key not in non_label_column_names
+                     and key != 'label'}
+        data_lengths = {len(array) for array in self.data.values()}
+        assert len(data_lengths) == 1  # Assert that all arrays have the same length
```
As a general style observation, we prefer not to have empty asserts in the code base (applies below as well)
```suggestion
        assert len(data_lengths) == 1, "Error message"  # Assert that all arrays have the same length
```

LysandreJik commented on examples/text-classification/new_tf_text_classification.py:
```diff
@@ -0,0 +1,498 @@
+logger = logging.getLogger(__name__)
+
+# region Helper classes
+class DataSequence(tf.keras.utils.Sequence):
```
Maybe too complex for this example, but I believe it's generally best to have the longest sequence be part of the initial batch, as the full batch then has the maximum (`seq_len x batch_size`) that will be encountered during training; if the training was to OOM at some point, it would do so right off the bat rather than 1hr in.

Unless I'm mistaken, that's not done here, would it be easy to add?

LysandreJik commented on examples/text-classification/new_tf_text_classification.py:
```diff
@@ -0,0 +1,498 @@
+        default=None,
+        metadata={"help": "Where do you want to store the pretrained models downloaded from huggingface.co"},
+    )
+    use_fast_tokenizer: bool = field(
+        default=True,
+        metadata={"help": "Whether to use one of the fast tokenizer (backed by the tokenizers library) or not."},
+    )
```
I think we should just always recommend a fast tokenizer, so I'm not even sure we really need that here. Thoughts @sgugger?

sgugger replied:
Yes, this predates v4 when we had to actually set `use_fast=True`, we can probably remove that arg now.

Rocketknight1 replied:
Done!

LysandreJik commented on examples/text-classification/new_tf_text_classification.py:
```diff
@@ -0,0 +1,498 @@
+        revision=model_args.model_revision,
+        use_auth_token=True if model_args.use_auth_token else None,
+    )
+    optimizer = tf.keras.optimizers.Adam(
+        learning_rate=training_args.learning_rate,
+        beta_1=training_args.adam_beta1,
+        beta_2=training_args.adam_beta2,
+        epsilon=training_args.adam_epsilon,
+        clipnorm=training_args.max_grad_norm)
+    if is_regression:
+        loss = tf.keras.losses.MeanSquaredError()
+    else:
+        loss = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+    model.compile(optimizer=optimizer, loss=loss)
```
I wouldn't put this in `# region Load pretrained model and tokenizer`, but maybe something like `# region Define optimizer, loss and compilation`? I feel like this is something that should be separate

Rocketknight1 replied:
Done!

LysandreJik reviewed (commented):
I find this wonderfully simple to get through, great job @Rocketknight1!

I left a few comments below.

sgugger commented on examples/text-classification/new_tf_text_classification.py:
```diff
@@ -0,0 +1,498 @@
+    # Detecting last checkpoint.
+    checkpoint = None
+    if len(os.listdir(training_args.output_dir)) > 0 and not training_args.overwrite_output_dir:
+        if (output_dir / 'config.json').is_file() and (output_dir / 'tf_model.h5').is_file():
```
Nit: we should import the constant containing the names from transformers.file_utils.

Rocketknight1 replied:
Done!

sgugger commented on examples/text-classification/new_tf_text_classification.py:
```diff
@@ -0,0 +1,498 @@
+        for index in random.sample(range(len(train_dataset)), 3):
+            logger.info(f"Sample {index} of the training set: {train_dataset[index]}.")
+
+    if data_args.pad_to_max_length:
+        data_collator = default_data_collator
+    elif training_args.fp16:
+        data_collator = DataCollatorWithPadding(tokenizer, pad_to_multiple_of=8)
+    else:
+        data_collator = None
```
This is PyTorch specific and should be removed.

Rocketknight1 replied:
Done!

sgugger commented on examples/text-classification/new_tf_text_classification.py:
```diff
@@ -0,0 +1,498 @@
+
+    # region Training
+    if training_args.do_train:
+        # checkpoint = None
+        # if last_checkpoint is not None:
+        #     checkpoint = last_checkpoint
+        # elif os.path.isdir(model_args.model_name_or_path):
+        #     # Check the config from that potential checkpoint has the right number of labels before using it as a
+        #     # checkpoint.
+        #     if AutoConfig.from_pretrained(model_args.model_name_or_path).num_labels == num_labels:
+        #         checkpoint = model_args.model_name_or_path
+        #
+        # if last_checkpoint is not None:
+        #     model.load_weights(last_checkpoint)
```
I think we can clean this up for now.

Rocketknight1 replied:
Done!

sgugger reviewed (approved):
Looking great! Thanks a lot for working on this!

sgugger committed:
Trainer iterable dataset (#11254)

* IterableDatasetShard

* Test and integration in Trainer

* Update src/transformers/trainer_pt_utils.py

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

* Style

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

Narsil committed:
Adding pipeline task aliases. (#11247)

* Adding task aliases and adding `token-classification` and
`text-classification` tasks.

* Cleaning docstring.

Rocketknight1 committed:
Update copyright date

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Rocketknight1 committed:
Adding error messages to asserts

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

sgugger committed:
Support for set_epoch (#11258)

Rocketknight1 commented on examples/text-classification/new_tf_text_classification.py:
```diff
@@ -0,0 +1,498 @@
+    def __getitem__(self, item):
+        # Note that this yields a batch, not a single sample
+        batch_start = item * self.batch_size
+        batch_end = (item + 1) * self.batch_size
```
We don't need it, though we could put it in for clarity! If you pass slice indices past the end of a list/array, you just get a slice up to the end of the list/array, e.g.
```
>>> mylist = [1, 2, 3, 4, 5]
>>> mylist[3:8]
[4, 5]
>>>
```

Rocketknight1 committed:
Cleaning up unused sections, adding comment for the callback

Rocketknight1 committed:
Merge branch 'tf_examples' of github.com:huggingface/transformers into tf_examples

Rocketknight1 committed:
Added meaningful assert messages, removed use_fast args

Rocketknight1 commented on examples/text-classification/new_tf_text_classification.py:
```diff
@@ -0,0 +1,498 @@
+    TFAutoModelForSequenceClassification,
+    AutoTokenizer,
+    DataCollatorWithPadding,
+    EvalPrediction,
```
Done!

Rocketknight1 committed:
Import filenames from file_utils

Rocketknight1 committed:
Separated optimizer, loss and compilation into their own region

Rocketknight1 committed:
Black formatting

Rocketknight1 committed:
Import sorting

Rocketknight1 committed:
Code quality cleanups

sgugger committed:
Tokenizer fast save (#11234)

* Save fast tokenizers in both formats

* Fix for HerBERT

* Proper fix

* Properly test new behavior

Rocketknight1 committed:
Code quality cleanups

Rocketknight1 committed:
Flake8 cleanups

Rocketknight1 committed:
Cleanup on comments and logging messages

stas00 committed:
update dependency_versions_table (#11273)

missed this updating when bumped the version.

LysandreJik committed:
Workflow fixes (#11270)

Narsil committed:
Enabling multilingual models for translation pipelines. (#10536)

* [WIP] Enabling multilingual models for translation pipelines.

* decoder_input_ids -> forced_bos_token_id

* Improve docstring.

* Rebase

* Fixing 2 bugs

- Type token_ids coming from `_parse_and_tokenize`
- Wrong index from tgt_lang.

* Fixing black version.

* Adding tests for _build_translation_inputs and add them for all
tokenizers.

* Mbart actually puts the lang code at the end.

* Fixing m2m100.

* Adding TF support to `deep_round`.

* Update src/transformers/pipelines/text2text_generation.py

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Adding one line comment.

* Fixing M2M100 `_build_translation_input_ids`, and fix the call site.

* Fixing tests + deep_round -> nested_simplify

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

LysandreJik committed:
Fix failing workflows

Rocketknight1 committed:
Writing README and starting the new folder structure
for TF examples

Rocketknight1 committed:
Writing README and starting the new folder structure
for TF examples

sgugger commented on examples/tensorflow/text-classification/README.md:
```diff
@@ -0,0 +1,65 @@
+
+# Text classification examples
+
+This folder contains some scripts showing examples of *text classification* with the Transformers library.
```
```suggestion
This folder contains some scripts showing examples of *text classification* with the 🤗 Transformers library.
```

sgugger commented on examples/tensorflow/text-classification/requirements.txt:
```diff
@@ -0,0 +1,3 @@
+datasets >= 1.1.3
+sentencepiece != 0.1.92
+protobuf
```
Add tensorflow here :-)

Rocketknight1 replied:
Done!

Rocketknight1 replied:
Done!

sgugger reviewed (commented):
I like the new README!

Rocketknight1 committed:
Remove the need for --do_train, --do_eval and --do_test,
Correctly write id2label to the output config
so we can make predictions

sgugger committed:
Trainer support for IterableDataset for evaluation and predict (#11286)

* Bulk of the work

* Polish and tests

* Update QA Trainer

* Avoid breaking the predict method

* Deprecation warnings

* Store real eval dataloder

* Get eval dataset reference before wrap

Commit:
move device statements outside if statements (#11292)

taepd committed:
modify double considering special tokens in `language_modeling.py` (#11275)

* Update language_modeling.py

in "class TextDatasetForNextSentencePrediction(Dataset)", double considering "self.tokenizer.num_special_tokens_to_add(pair=True)" 

so, i remove self.block_size, and add parameter for "def create_examples_from_document". like "class LineByLineWithSOPTextDataset" do

* Update language_modeling.py

stas00 committed:
[Trainer] fix the placement on device with fp16_full_eval (#11322)

* fix the placement on device with fp16_full_eval

* deepspeed never goes on device

sgugger committed:
[Trainer] Add a progress bar for batches skipped (#11324)

sgugger committed:
Load checkpoint without re-creating the model (#11318)

rajvi-k committed:
Added translation example script  (#11196)

* initial changes

* modified evaluation

* updated evaluation

* updated evaluation on text translation example script

* added translation example script

* Formatted translation example script

* Reformatted translation example

* Fixed evaluation bug and added support for other tokenisers

* Fixed evaluation bug and added support for other tokenisers

* Added translation example script

* Formatted summarization example script

* Removed typos from summarization example script

Rocketknight1 committed:
Making sure everything aligns between pretrained models and future data

patrickvonplaten committed:
[Generate] Remove outdated code (#11331)

* remove update function

* update

* refactor more

* refactor

patil-suraj committed:
[GPTNeo] create local attention mask ones (#11335)

* create local attention mask ones

* remove old method, address patricks comment

Rocketknight1 committed:
Example script looks ready to go!
Next step is multi-GPU and TPU training

sgugger reviewed (approved):
LGTM, left a few comments, mostly about formatting, and I think we could add accuracy as a metric if we don't have a regression task.

sgugger committed:
Update to use datasets remove_cloumns method (#11343)

* Update to use datasets remove_cloumns method

* Quality

forest1988 committed:
Add an error message that fires when Reformer is not in training mode, but one runs .backward() (#11117)

Narsil committed:
Removed `max_length` from being mandatory within `generate`. (#11314)

* Removed `max_length` from being mandatory within `generate`.

- Moving on to fully using `StoppingCriteria` for `greedy` and `sample`
modes.
- `max_length` still used for `beam_search` and `group_beam_search`
(Follow up PR)
- Fixes a bug with MaxLengthStoppingCriteria (we should stop as soon a
we hit the max_length, the comparison needs to be or equal, that affects
the tests).
- Added options to use `logits_processor` and `stopping_criteria`
directly within `generate` function (so some users can define their own
`logits_processor` and `stopping_criteria`).
- Modified the backward compat tests to make sure we issue a warning.

* Fix `max_length` argument in `generate`.

* Moving validate to being functional.

- Renamed `smax_length` to `stoppping_max_length`.

* Removing `logits_processor` and `stopping_criteria` from `generate`
arguments.

* Deepcopy.

* Fix global variable name.

Rocketknight1 committed:
Formatting/code quality

Rocketknight1 committed:
Update examples/tensorflow/text-classification/README.md

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Rocketknight1 commented on examples/tensorflow/text-classification/run_text_classification.py:
```diff
@@ -0,0 +1,537 @@
+    validation_file: Optional[str] = field(default=None,
+                                           metadata={"help": "A csv or a json file containing the validation data."})
+    test_file: Optional[str] = field(default=None,
+                                     metadata={"help": "A csv or a json file containing the test data."})
```
I forgot the formatting run again, sorry. Fixed!

Rocketknight1 replied:
Done!

Rocketknight1 commented on examples/tensorflow/text-classification/run_text_classification.py:
```diff
@@ -0,0 +1,537 @@
+    )
+    logger.setLevel(logging.INFO)
+
+    # Set the verbosity to info of the Transformers logger (on main process only):
+    if is_main_process(training_args.local_rank):
+        transformers.utils.logging.set_verbosity_info()
+        transformers.utils.logging.enable_default_handler()
+        transformers.utils.logging.enable_explicit_format()
```
Done!

Rocketknight1 replied:
Done!

Rocketknight1 committed:
Remove mostly Pytorch-related logging

Rocketknight1 committed:
Merge remote-tracking branch 'origin/tf_examples' into tf_examples

Rocketknight1 committed:
Update requirements.txt

Rocketknight1 commented on examples/tensorflow/text-classification/run_text_classification.py:
```diff
@@ -0,0 +1,537 @@
+    # download the dataset.
+    data_files = {"train": data_args.train_file,
+                  "validation": data_args.validation_file,
+                  "test": data_args.test_file}
```
Done!

Rocketknight1 commented on examples/tensorflow/text-classification/run_text_classification.py:
```diff
@@ -0,0 +1,537 @@
+        loss = tf.keras.losses.MeanSquaredError()
+    else:
+        loss = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+    model.compile(optimizer=optimizer, loss=loss)
```
Done!

Rocketknight1 committed:
Support passing only a validation file,
add accuracy as a metric when we're doing classification

Rocketknight1 commented on examples/tensorflow/text-classification/run_text_classification.py:
```diff
@@ -0,0 +1,537 @@
+        eval_dataset = DataSequence(
+            eval_dataset, non_label_column_names, batch_size=training_args.per_device_eval_batch_size, labels=True
+        )
+        print("Computing loss on validation data...")
```
Done!

sgugger committed:
Honor contributors to models (#11329)

* Honor contributors to models

* Fix typo

* Address review comments

* Add more authors

Rocketknight1 committed:
Formatting pass

Rocketknight1 committed:
Remove unused imports

Rocketknight1 committed:
Minor cleanup of a duplicate assertion

sgugger commented on examples/tensorflow/text-classification/run_text_classification.py:
```diff
@@ -0,0 +1,532 @@
+        eval_dataset = DataSequence(
+            eval_dataset, non_label_column_names, batch_size=training_args.per_device_eval_batch_size, labels=True
+        )
+        logger.info("Computing metrics  on validation data...")
```
```suggestion
        logger.info("Computing metrics on validation data...")
```

sgugger reviewed (approved):
Great work!

Rocketknight1 committed:
Update examples/tensorflow/text-classification/run_text_classification.py

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

LysandreJik reviewed (approved):
LGTM, great work @Rocketknight1!

philschmid commented on examples/tensorflow/text-classification/run_text_classification.py:
```diff
@@ -0,0 +1,532 @@
+            eval_dataset = None
+
+        callbacks = [SavePretrainedCallback(output_dir=training_args.output_dir)]
+        model.fit(training_dataset, validation_data=eval_dataset, epochs=10, callbacks=callbacks)
```
Is it on purpose that `epochs=10` is hardcoded? shouldn't this be `epochs=training_args. num_train_epochs`?

Rocketknight1 replied:
Aagh, you're totally right. Fixing it!

philschmid reviewed (commented):
Hey just took a look at your the `TF` example it looks amazing. I noticed one thing what seemed strange to me

Rocketknight1 committed:
Stopped hardcoding the number of epochs

Rocketknight1 committed:
Merge remote-tracking branch 'origin/tf_examples' into tf_examples

stas00 committed:
[deepspeed] fix resume from checkpoint (#11352)

This PR fixes a bug that most likely somehow got exposed (not caused) by https://github.com/huggingface/transformers/pull/11318 - surprisingly the same test worked just fine before that other PR.

sgugger committed:
Examples reorg (#11350)

* Base move

* Examples reorganization

* Update references

* Put back test data

* Move conftest

* More fixes

* Move test data to test fixtures

* Update path

* Apply suggestions from code review

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

* Address review comments and clean

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

lewtun committed:
Extract metric_key_prefix during NotebookProgressCallback.on_evaluate (#11347)

* Pass metric_key_prefix as kwarg to on_evaluate

* Replace eval_loss with metric_key_prefix_loss

* Default to "eval" if metric_key_prefix not in kwargs

* Add kwargs to CallbackHandler.on_evaluate signature

* Revert "Add kwargs to CallbackHandler.on_evaluate signature"

This reverts commit 8d4c85ed512f558f7579d36771e907b3379947b7.

* Revert "Pass metric_key_prefix as kwarg to on_evaluate"

This reverts commit 7766bfe2718601230ae593d37b1317bd53cfc075.

* Extract metric_key_prefix from metrics

Rocketknight1 committed:
Merge branch 'master' into tf_examples

# Conflicts:
#	examples/tensorflow/text-classification/README.md
#	examples/tensorflow/text-classification/requirements.txt

Rocketknight1 commented:
I opened a new branch and PR at #11360 to avoid dealing with rebasing after the folder structure was changed around
